java/src/main/java/com/kevinthorley/csi280/project01/ClassRepo.java

```diff
@@ -14,12 +14,16 @@ public ClassRepo() {
     this.sections.add(new Section("csi-280-52", "Open Source Software Development", "2016SP"));
   }
 
-  public Section findById(String id) {
-    // TODO: find a section by its id
-    return null;
-  }
+public Section findById(String id) {
+
+    for (int i = 0; i < this.sections.size(); i++) {
+	    if (id == Section.id){
+	      return Section;
+    	}
+      return null;
+    }
```

Comment:
The overall structure was correct. Here is how I modified your code to get it to compile:

`for (int i = 0; i < this.sections.size(); i++) {
            if (id == this.sections.get(i).getId()){
              return this.sections.get(i);
        }
    }

    return null;`